doc/rados/operations/crush-map-edits.rst

```diff
@@ -32,193 +32,204 @@ Pool Values`_.
+   particular pool.
 
 
 .. _crushmapdevices:
 
-CRUSH Map Devices
+CRUSH-Map Devices
 -----------------
 
-Devices are individual OSDs that store data.  Usually one is defined here for each
-OSD daemon in your
-cluster.  Devices are identified by an ``id`` (a non-negative integer) and
-a ``name``, normally ``osd.N`` where ``N`` is the device id.
+Devices are individual OSDs that store data. In this section, there is usually
+one device defined for each OSD daemon in your cluster. Devices are identified
+by an ``id`` (a non-negative integer) and a ``name`` (usually ``osd.N``, where
+``N`` is the device's ``id``).
+
 
 .. _crush-map-device-class:
 
-Devices may also have a *device class* associated with them (e.g.,
-``hdd`` or ``ssd``), allowing them to be conveniently targeted by a
-crush rule.
+A device can also have a *device class* associated with it: for example,
+``hdd`` or ``ssd``. Device classes make it possible for devices to be targeted
+by CRUSH rules. This means that device classes allow CRUSH rules to select only
+OSDs that match certain characteristics. For example, you might want an RBD
+pool associated only with SSDs and a different RBD pool associated only with
+HDDs.
+
+To see a list of devices, run the following command:
 
 .. prompt:: bash #
 
-	devices
+   ceph device ls 
+
+The output of this command takes the following form:
 
 ::
 
-	device {num} {osd.name} [class {class}]
+    device {num} {osd.name} [class {class}]
 
 For example:
 
 .. prompt:: bash #
 
-	devices
+    ceph device ls
 
 ::
 
-	device 0 osd.0 class ssd
-	device 1 osd.1 class hdd
-	device 2 osd.2
-	device 3 osd.3
+    device 0 osd.0 class ssd
+    device 1 osd.1 class hdd
+    device 2 osd.2
+    device 3 osd.3
+
+In most cases, each device maps to a corresponding ``ceph-osd`` daemon. This
+daemon might map to a single storage device, a pair of devices (for example,
+one for data and one for a journal or metadata), or in some cases a small RAID
+device or a partition of a larger storage device.
 
-In most cases, each device maps to a single ``ceph-osd`` daemon.  This
-is normally a single storage device, a pair of devices (for example,
-one for data and one for a journal or metadata), or in some cases a
-small RAID device.
 
-CRUSH Map Bucket Types
+CRUSH-Map Bucket Types
 ----------------------
 
-The second list in the CRUSH map defines 'bucket' types. Buckets facilitate
-a hierarchy of nodes and leaves. Node (or non-leaf) buckets typically represent
-physical locations in a hierarchy. Nodes aggregate other nodes or leaves.
-Leaf buckets represent ``ceph-osd`` daemons and their corresponding storage
-media.
+The second list in the CRUSH map defines 'bucket' types. Buckets facilitate a
+hierarchy of nodes and leaves. Node buckets (also known as non-leaf buckets)
+typically represent physical locations in a hierarchy. Nodes aggregate other
+nodes or leaves.  Leaf buckets represent ``ceph-osd`` daemons and their
```

Comment:
@zdover23 - A non-puzzling introduction of double spaces (it's the product of a sentence beginning on a new line).

Reply:
Fixed.